config/core/configmaps/gc.yaml

```diff
@@ -60,27 +60,52 @@ data:
+    # These settings are enabled via the "responsive-revision-gc" feature flag.
     # ALPHA NOTE: This feature is still experimental and under active development.
+    #
+    # Stale
+    #   * Revisions are considered stale if they were created after more than
+    #      "retain-since-create-time" or last referenced by a route more than
+    #      "retain-since-last-active-time" ago.
+    #   * Either timeout may be set to "disabled" to turn off time-based GC.
+    # Active
+    #   * Revisions which are referenced by a Route are considered active and
+    #     are exempted from GC.
+    #   * Individual revisions may be marked with the annotation
+    #      "knative.dev/no-gc":"true" to be permanently considered active.
+    #
+    # Example config to immediately collect any inactive revision:
+    #    min-stale-revisions: "0"
+    #    retain-since-create-time: "0"
+    #    retain-since-last-active-time: "0"
+    #
+    # Example config to always keep around the last ten revisions:
+    #     retain-since-create-time: "disabled"
+    #     retain-since-last-active-time: "disabled"
+    #     max-non-active-revisions: "10"
+    #
+    # Example disable all GC:
+    #     retain-since-create-time: "disabled"
+    #     retain-since-last-active-time: "disabled"
+    #     max-non-active-revisions: "disabled"
+    #
+    # Example config to keep recently deployed or active revisions,
+    # always maintain the last two in case of rollback, and prevent
+    # burst activity from exploding the count of old revisions:
+    #      retain-since-create-time: "48h"
+    #      retain-since-last-active-time: "15h"
+    #      min-stale-revisions: "2"
+    #      max-non-active-revisions: "1000"
 
-    # Duration since revision creation before considering it for GC.
+    # Duration since creation before considering a revision for GC or "disabled".
     retain-since-create-time: "48h"
 
-    # Duration since a route has pointed at the revision
-    # before considering it for GC.
+    # Duration since active before considering a revision for GC or "disabled".
     retain-since-last-active-time: "15h"
 
-    # Minimum number of non-active revisions to retain.
+    # Minimum number of stale revisions to retain.
     min-stale-revisions: "20"
 
-    # Maximum number of non-active revisions to retain regardless of
-    # creation or staleness time-bounds.
-    #
-    # Instead of allowing revisions to become stale down to min-stale-revisions,
-    # this mode will keep around revisions up to this maximum limit. A value of 0
-    # will immediately delete any non-active revisions. The expected total number
-    # of revisions is the number of active plus this value.
-    #
-    # Set to -1 to disable this setting. Revisions will be eligible for gc as soon
-    # as they are considered stale as long as there are more than min-stale-revisions.
-    max-stale-revisions: "-1"
+    # Maximum number of non-active revisions to retain
```

Comment:
So why is this one `non-active` rather than stale?

Reply:
this is a total limit on non-active and ignores staleness. If you create a bunch and they're non-active but not stale yet either, this still limits your history of revisions at a hard count.

Reply:
It's subtle. I am afraid will lead to user confusion :)

Reply:
Completely agree. That's why I included the examples of common modes, but I wish it were both configurable and more natural by-design. Open to suggestions.

Reply:
Probably good docs with example would be helpful for this. You can probably take it to docs WG meeting to discuss how to best write this out.

Reply:
I'm removing the word 'stale'. Both refer to bounds on non-active revisions.